src/smc-hub/postgres/set-pg-params.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ *  This file is part of CoCalc: Copyright © 2020 Sagemath, Inc.
+ *  License: AGPLv3 s.t. "Commons Clause" – see LICENSE.md for details
+ */
+
+// This is used in postgres-base to set postgres parameters for a single query
+// https://www.postgresql.org/docs/10/sql-set.html
+
+import { Client } from "pg";
+import * as debug from "debug";
+const L = debug("hub:pg:query_params");
+
+interface Opts {
+  client: Client;
+  query: string;
+  params: string[];
+  pg_params: { [key: string]: string };
+  cb: (err?, result?) => void;
+}
+
+// run the actualy query after a setup quey in a transaction
+// used by __do_query in postgres-base
+export async function do_query_with_pg_params(opts: Opts): Promise<void> {
+  const { client, query, params, pg_params, cb } = opts;
+
+  try {
+    await client.query("BEGIN");
+    for (const [k, v] of Object.entries(pg_params)) {
+      // SET LOCAL: only for this transaction!
+      // NOTE: interestingly, $1, $2 params do not work … but this isn't user-facing
+      const q = `SET LOCAL ${k} TO ${v}`;
+      await client.query(q);
+    }
+    const res = await client.query(query, params);
+    cb(undefined, res);
```

Comment:
IMPORTANT: Lines 35 and 36 should be swapped.  The cb(undefined, res) must happen after the await.  Otherwise, if `await client.query("COMMIT")` fails (likely!), then cb will be called _twice_, once with no error and once with an error; not good.